staging/src/k8s.io/client-go/discovery/cached/disk/cached_discovery_test.go

```diff
@@ -207,3 +289,7 @@ func (c *fakeDiscoveryClient) OpenAPISchema() (*openapi_v2.Document, error) {
 	c.openAPICalls = c.openAPICalls + 1
 	return &openapi_v2.Document{}, nil
 }
+
+func (d *fakeDiscoveryClient) OpenAPIV3() openapi.Client {
+	panic("unimplemented")
```

Comment:
There are multiple fake discovery clients which appear to exist to test different aspects of the discovery API, or created by different people who didn't know about the old ones. 

A side effect of all these different mock discovery client implementations is that if I add a method to discovery interface, I must add a stub to all the implementations, even if my interface isn't used in their respective tests. 

https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/restmapper/shortcut_test.go#L230

https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/restmapper/discovery_test.go#L420

https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/discovery/cached/disk/cached_discovery_test.go#L126

https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/discovery/fake/discovery.go#L36

Unfortunately I don't have the bandwidth to untangle these here in this PR.

Addressing another one of your comments here since it is related; for my own openapiv3 tests I did not use any of these mock clients and instead used Go's `httptest.server` to test against a "real" openapi local server: `https://github.com/alexzielenski/kubernetes/blob/cache-busting-client-go/staging/src/k8s.io/client-go/util/testing/fake_openapi_handler.go#L47`. This server is easy/portable enough for others to test their use of the openapiV3 client.

Reply:
OK